features/discovery/src/main/java/org/opennms/netmgt/discovery/UnmanagedInterfaceFilter.java

```diff
@@ -50,7 +48,7 @@ public UnmanagedInterfaceFilter(InterfaceToNodeCache interfaceToNodeCache) {
 
     @Override
     public boolean matches(String location, InetAddress address) {
-        return Iterables.isEmpty(interfaceToNodeCache.getNodeId(location, address));
+        return interfaceToNodeCache.getFirstNodeId(location, address).isEmpty();
```

Comment:
I do not see how it can removed without changing the API layer.

Reply:
That's exactly why I'm asking. The API layer is not stable yet - but it will be soon. Therefore this is the right time to remove such stuff if it is not needed.

Reply:
@RangerRick @cgorantla Do you have an opinion on `org.opennms.integration.api.v1.dao.InterfaceToNodeCache`? There is the method

```
Stream<Integer> getNodeIds(String location, InetAddress ipAddr);
```

Can that method be removed? In that case the underlying implementation could get rid of the corresponding method.

Reply:
Nothing appears to be using it now, so good to remove.

It was added because I thought it may be useful, but we can revisit this once we have a good use case for it.